src/Dispatcher/Result.php

```diff
@@ -0,0 +1,185 @@
+<?php
+declare(strict_types=1);
+
+namespace FastRoute\Dispatcher;
+
+use FastRoute\RouteInterface;
+use RuntimeException;
+
+/**
+ * Result Object
+ */
+class Result implements ResultInterface
+{
+    public const NOT_FOUND = 0;
+    public const FOUND = 1;
+    public const METHOD_NOT_ALLOWED = 2;
+
+    protected bool $matched = false;
+
+    protected ?RouteInterface $route;
```

Comment:
I don't think this should be here since this will bring some unnecessary object reference and doesn't provide much value from a dispatching perspective.